Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
```

Comment:
As I understand it, the goal should be:
- Maintain a pool of connections that live longer that server HTTP requests. This is, new incoming requests can grab a connection from the pool, execute multiple queries/transactions and finally return the connection to the pool for reuse by upcoming requests.
- While a request (thread) is using a connection, no other request (thread) should be able to use it.

This can be achieved by having two methods to:
1. Capture a connection from the pool (or create a new one if pool is empty).
2. Return the connection to the pool when request is complete.

Current implementation might work, but connections created by a thread are not reusable by upcoming requests, unless the thread id did match, which would rarely ever happen. This means each request/thread will need to create a new, non-reusable connection, similar to what happens right now.